std/socket.d

```diff
@@ -290,30 +290,34 @@ bool wouldHaveBlocked() nothrow @nogc
 }
 
 
-private immutable
+version (Windows)
 {
+    private: immutable:
     typeof(&getnameinfo) getnameinfoPointer;
     typeof(&getaddrinfo) getaddrinfoPointer;
     typeof(&freeaddrinfo) freeaddrinfoPointer;
 }
+else version (Posix)
+{
+    enum getnameinfoPointer = &getnameinfo;
+    enum getaddrinfoPointer = &getaddrinfo;
+    enum freeaddrinfoPointer = &freeaddrinfo;
```

Comment:
Remove these for `Posix`, add `version(Windows) { ... } else version(Posix) { ... } else static assert(0);` where applicable to avoid un-necessary indirections.

Reply:
Thanks, I'll add the assert. To get rid of the enums I'd need to wait for the updated Windows version.

Reply:
Update: I see there's a `static assert` on line 126, so no need to duplicate that.

Reply:
Just checked, only dmd generates actual indirections :). https://godbolt.org/z/RmDYYV

Reply:
Even better: I consolidated those to the top version blocks.